### ISSUE TITLE
Custom cover image?

### ISSUE THREAD

**Issue by michaeltieso** [paulirish/lite-youtube-embed#64]
Thanks for this!
Is there a way to insert your own cover image for a video?

**Comment by Garbee** [comment]
This feels like a duplicate to #54 which aims to discuss a solution that would allow for this as well.

Currently it is not supported since we automatically pull the cover image generated by YouTube. To be honest, I'd urge against supporting this directly. As it could be very easily mis-used to trick people into viewing video's they don't expect.

What exactly is your use-case for wanting a fully custom cover image?

**Comment by michaeltieso** [comment]
Good question. The thumbnail provided by YouTube is low quality. While the cover image itself where I'm using this will be the same, the the low resolution of the image makes for a blurry image and would prefer to implement an image of higher resolution.

> To be honest, I'd urge against supporting this directly. As it could be very easily mis-used to trick people into viewing video's they don't expect.

I understand but don't think that's something you can control as that's already possible via YouTube thumbnail. I also don't think it's something we could/should police.

**Comment by Fiorinol** [comment]
Same issue here. YouTube thumbnails are of low quality. I'm currently looking for another library that does what lite-youtube does, but without the image limitations/being unable to lazyload images out of the box.

**Comment by davidsneighbour** [comment]
The background image in the "pro usage" section of the readme should be your cover image?

https://github.com/paulirish/lite-youtube-embed#pro-usage

**Comment by IORoot** [comment]
Putting a conditional around the backgroundImage seems simplest, no?

```javascript
if (!this.style.backgroundImage){
    this.posterUrl = `https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg`;
    LiteYTEmbed.addPrefetch('preload', this.posterUrl, 'image');
    this.style.backgroundImage = `url("${this.posterUrl}")`;
}
```
If there's already a background image, leave it alone...

**Comment by philwolstenholme** [comment]
+1 for IORoot's suggestion, I've just been bitten by this a project where we let CMS users add custom cover images to videos (often higher quality ones than provided by YouTube) and this component is overwriting the custom image we are setting as a background image on the component.

**Comment by paulirish** [comment]
shipped in 0.2.0.  

i added docs and a demo, too: https://github.com/paulirish/lite-youtube-embed#custom-poster-image
